### DISCUSSION TITLE
Generation of pie charts

### DISCUSSION THREAD

**Discussion by RiccLQL** [excalidraw/excalidraw#4253] [Q&A]
Hi!

I'm absolutely amazed by the copy/paste charting feature, and my girlfriend uses it to generate quick charts for presentations. I wanted to help her by creating the option to also generate pie charts.

I can implement it but based on past discussions it looks like there are several challenges (i.e. unique colors). I was wondering if anyone already has tips, ideas and ways to build these pie charts - I think it would be a useful chart option to add:)

**Comment by pshihn** [comment]
As far as roughjs is concerned for rendering, i'd say use the `arc` method which would be ideal for this rather than compute your own paths (`arc (x, y, width, height, start, stop, closed [, options])`)

for example (codepen: https://codepen.io/shihn/pen/OJjBaOd): 
```javascript
rc.arc(350, 300, 200, 200, Math.PI, Math.PI * 1.6, true);
rc.arc(350, 300, 200, 200, 0, Math.PI / 2, true, {
  fill: 'rgba(255,255,0,0.4)',
  fillStyle: 'solid'
});
rc.arc(350, 300, 200, 200, Math.PI / 2, Math.PI, true, {
  fill: 'rgba(255,0,255,0.4)'
});
rc.arc(350, 300, 200, 200, Math.PI * 1.6, Math.PI * 2, true, {
  fill: 'orange',
  fillStyle: 'cross-hatch'
});
```
will render:

![Screen Shot 2021-11-14 at 1 18 51 PM](https://user-images.githubusercontent.com/833927/141699007-ac6eff2d-9d44-4fb5-9ccd-7d7d1d44f762.png)

Regarding selection of colors, the simplest solution would be to cycle through a standard palette. 
D3 has some scales defined here: https://github.com/d3/d3-scale-chromatic#categorical which work great for charts.

**Reply by RiccLQL** [reply]
Thanks! That helps:) As for colors, would you prefer I cycle through a palette of 8-16 colors and repeat, or I cycle through a continuous palette and pick random unique colors for every portion (thinking about the case where someone generates like 30 portions)? Considering at least computational speed / visual clarity / what is usually done for pie charts with lot of portions.

**Comment by RiccLQL** [comment]
As well, should I create an issue for this feature request?

**Reply by dwelle** [reply]
Yeah, let's open an issue for this.

**Reply by PeterBocan** [reply]
hi, did you manage to open an issue for this?
